### ISSUE TITLE
Unable to create MR: mrList.go:38: gitlab project not found

### ISSUE THREAD

**Issue by mwoodson** [zaquestion/lab#130]
I'm looking for some help with this.  I don't know what I'm doing wrong.

I have setup gitlab in a docker container.  I am forwarding into this docker container via apache and a proxy.  I am also running on a non-standard ssh port for git.

```
$ lab mr list     
2018/04/11 16:46:11 mrList.go:38: gitlab project not found
```

```
$ git remote -v
origin	ssh://git@git.mydomain.net:10022/mwoodson/repo.git (fetch)
origin	ssh://git@git.mydomain.net:10022/mwoodson/repo.git (push)
upstream	ssh://git@git.mydomain.net:10022/Project/repo.git (fetch)
upstream	ssh://git@git.mydomain.net:10022/Project/repo.git (push)
```

Any help would be greatly appreciated!

**Comment by zaquestion** [comment]
Hi @mwoodson #114 was a similar issue where the the user was running apache and having issues where apache was double encoding the urls. The fix for that was adding (below) to the config.
```
    AllowEncodedSlashes NoDecode
```

The issue also has all of our debugging steps in it which might be of help here. Addtionally I've added tests to assert that project names are correctly parsed with custom ssh ports in #131

**Comment by mwoodson** [comment]
@zaquestion thanks! I will try to post more debugging.

**Comment by mwoodson** [comment]
@zaquestion 

Thanks. This took me down a crazy path, and I'm not sure why others haven't experienced this.

Thanks to some tcpdump'ing, I saw what was happening.

`curl --header "Private-Token:  SECRET" 'https://git.mydomain.net/api/v4/projects/Project%2Frepo'`

The URL was getting re-written to:

`GET /api/v4/projects/Project%252Frepo HTTP/1.1\r\n`

Notice that '%2F' got re-encoded to '%252F'.  

To be clear, I had the option enabled in Apache:

`  AllowEncodedSlashes NoDecode
`

I was able to solve the issue with by adding the option "nocanon" to the end of the proxy pass line:

`ProxyPass / http://localhost:10080/ keepalive=On connectiontimeout=10 retry=0 nocanon`

Again, I'm not sure why other's aren't reporting this in the gitlab community, or if they are using a different setting to avoid this.

Things seem to be working.  Thanks for pushing me in the right direction.
